.github/workflows/golangci-lint.yaml

```diff
@@ -12,11 +12,13 @@ jobs:
     name: lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v4.0.1
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # tag=v3.5.3
+      - name: Calculate go version
+        id: vars
+        run: echo "go_version=$(make go-version)" >> $GITHUB_OUTPUT
+      - name: Set up Go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # tag=v4.0.1
         with:
-          go-version: 1.19
+          go-version: ${{ steps.vars.outputs.go_version }}
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.6.0
-        with:
-          version: v1.52.2
+        uses: golangci/golangci-lint-action@639cd343e1d3b897ff35927a75193d57cfcba299 # tag=v3.6.0
```

Comment:
I think we should still pin the golangci-lint version

Reply:
It's probably already part of what you're planning to do. Let's also get rid of the shell script to install golangci-lint and align to core CAPI

Reply:
Let me take bumping the linter out to a separate PR :-)

Reply:
Absolutely!